### PR TITLE
standalone: Fix Drupal7 password algorithm

### PR DESCRIPTION
**Pull request by artfulrobot**
Correct a copy-paste mistake from when the code was imported from Drupal 7.

Before: you can't login because your password is always "wrong"

After: your password is correctly checked!

### CODE REVIEW

**Comment by colemanw** [comment]
Thanks @artfulrobot! Just a couple stylechecker issues.

**Comment by ufundo** [comment]
Great find @artfulrobot !

Would like to merge this but as far as I can see this will invalidate passwords from existing Standalone installs - so I'm just wondering if there's any way we can manage the fallout from that?

If pre-existing passwords are unsalted and therefore less secure, is there any way we can check and reencrypt with the fixed algo?

**Comment by ufundo** [comment]
Merging after discussion with core team and a warning on [Standalone Mattermost channel](https://chat.civicrm.org/civicrm/channels/standalone)

Unblocks further work on D7 migration. If anyone wants to look at a parallel algos mitigation strategy between now and when this hits release I'd be happy to discuss further.

File: ext/standaloneusers/Civi/Standalone/PasswordAlgorithms/Drupal7.php [hunk 1]
```diff
@@ -168,7 +168,7 @@ protected function _d7_password_crypt($algo, $password, $setting) {
 
     // Convert the base 2 logarithm into an integer.
     $count = 1 << $count_log2;
-    $hash = hash($algo, $password, TRUE);
+    $hash = hash($algo, $salt . $password, TRUE);
```
Comment (ufundo) [inline ext/standaloneusers/Civi/Standalone/PasswordAlgorithms/Drupal7.php:171]: definitely needed a bit of salt before! :yum:

**Pull request merged**